dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIEJBManagedObjectFactoryImpl.java

```diff
@@ -89,28 +91,99 @@ public ManagedObjectContext createContext() {
         return managedObjectContext;
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     protected synchronized Bean<T> getBean() {
-        if (!this.ejbBeanLookupComplete) {
+        if (!ejbBeanLookupComplete) {
+           doInitialization();
+        }
+        return this.ejbBean;
+    }
+
+    private synchronized void doInitialization() {
+        if (!ejbBeanLookupComplete) {
+
+            List<EjbDescriptor<T>> partialMatchDescriptors = new LinkedList<EjbDescriptor<T>>();
 
             WeldManager beanManager = getBeanManager();
 
             if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
                 Tr.debug(tc, "Looking for EJB Bean: " + this.ejbName);
             }
 
-            EjbDescriptor<?> ejbDescriptor = getEjbDescriptor();
-            if (ejbDescriptor != null) {
-                this.ejbBean = (Bean<T>) beanManager.getBean(ejbDescriptor);
+            if (this.ejbDescriptor == null) {
```

Comment:
I think this `if` shouldn't be needed any more, this method should only be called if initialization hasn't been done yet and so `ejbDescriptor` will never be set at this point.